packages/nimble-components/src/select/index.ts

```diff
@@ -357,9 +342,38 @@ export class Select extends FormAssociatedSelect implements ErrorPattern {
      * @override
      */
     public override handleChange(source: unknown, propertyName: string): void {
-        super.handleChange(source, propertyName);
-        if (propertyName === 'value') {
-            this.updateValue();
+        // don't call super.handleChange so hidden options can be selected programmatically
+        const sourceElement = source as Element;
+        switch (propertyName) {
+            case 'value': {
+                this.updateValue();
+                break;
+            }
+            case 'selected': {
+                if (isListboxOption(sourceElement)) {
+                    this.selectedIndex = this.options.indexOf(sourceElement);
+                }
+                this.setSelectedOptions();
+                this.updateDisplayValue();
+                break;
+            }
+            case 'hidden': {
+                if (isListboxOption(sourceElement)) {
+                    if (sourceElement.hidden) {
+                        sourceElement.classList.add('hidden-option');
```

Comment:
[we don't own the class attribute / classList configuration of an element](https://github.com/ni/nimble/blob/33354140e84c85eff7fe47c2452f559a29a52ac0/packages/nimble-components/CONTRIBUTING.md#why-attributes-over-classes) but we do own attribute definition. To avoid potentially fighting with the user manipulating classes on the element can define an internal property or attribute?
And like we've chatted about before, all else equal when dealing with internal state a property is preferred but in this case since we want to style against it an attribute is perfectly reasonable.
Maybe `visually-hidden` (to distinguish from logically hidden)?